src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+					{
+						LastScrollView.ContentInset = StartingContentInsets;
+						LastScrollView.ScrollIndicatorInsets = StartingScrollIndicatorInsets;
+					}
+
+					var superScrollView = LastScrollView;
+					while (superScrollView is not null)
+					{
+						var contentSize = new CGSize(Math.Max(superScrollView.ContentSize.Width, superScrollView.Frame.Width),
+							Math.Max(superScrollView.ContentSize.Height, superScrollView.Frame.Height));
+
+						var minY = contentSize.Height - superScrollView.Frame.Height;
+						if (minY < superScrollView.ContentOffset.Y)
+						{
+							var newContentOffset = new CGPoint(superScrollView.ContentOffset.X, minY);
+							if (!superScrollView.ContentOffset.Equals(newContentOffset))
+							{
+								if (View?.Superview is UIStackView)
+									superScrollView.SetContentOffset(newContentOffset, UIView.AnimationsEnabled);
+								else
+									superScrollView.ContentOffset = newContentOffset;
+							}
+						}
+						superScrollView = superScrollView.Superview as UIScrollView;
+					}
+				});
+			}
+
+			if (IsKeyboardShowing)
+				RestorePosition();
+
+			IsKeyboardShowing = false;
+			View = null;
+			LastScrollView = null;
+			KeyboardFrame = CGRect.Empty;
+			StartingContentInsets = new UIEdgeInsets();
+			StartingScrollIndicatorInsets = new UIEdgeInsets();
+			StartingContentInsets = new UIEdgeInsets();
+		});
+	}
+
+	internal static void Destroy()
+	{
+		if (WillShowToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(WillShowToken);
+		if (DidHideToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(DidHideToken);
+		if (TextFieldToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextFieldToken);
+		if (TextViewToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextViewToken);
+	}
+
+	// Used to get the numeric values from the UserInfo dictionary's NSObject value to CGRect.
+	// Doing manually since CGRectFromString is not yet bound
+	static CGRect? DescriptionToCGRect(string description)
+	{
+		// example of passed in description: "NSRect: {{0, 586}, {430, 346}}"
+
+		if (description is null)
+			return null;
+
+		// remove letters in all languages, spaces, and curly brackets
+		var temp = Regex.Replace(description, @"[\p{L}\s:{}]", "");
+		var dimensions = temp.Split(',');
+
+		if (int.TryParse(dimensions[0], out var x) && int.TryParse(dimensions[1], out var y)
+			&& int.TryParse(dimensions[2], out var width) && int.TryParse(dimensions[3], out var height))
+		{
+			return new CGRect(x, y, width, height);
+		}
+
+		return null;
+	}
+
+	static async Task SetUpTextEdit()
+	{
+		if (View is null)
+			return;
+
+		CursorRect = null;
+
+		RootController = View.FindResponder<ContainerViewController>()?.View;
+
+		// the cursor needs a small amount of time to update the position
+		await Task.Delay(5);
+
+		UITextRange? selectedTextRange;
+		CGRect? localCursor = null;
```

Comment:
The default value of a nullable ref/value is null. You are initialising them to null several times, be consistent. Either we set it to null or not.

Reply:
If I don't set the localCursor = null, then I get this warning as error when I try to assign it
<img width="971" alt="image" src="https://user-images.githubusercontent.com/50846373/220818825-1a7a362f-6d12-4b16-bbeb-aff3c6452f00.png">